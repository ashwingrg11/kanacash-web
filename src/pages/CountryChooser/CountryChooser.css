/* .choose-country{
    border:1px solid rgb(226, 226, 226);
    border-radius: 10px;
    width: 100%;
    margin-top: 10px;
}



.country-search
{
    background: transparent;
    border: none;
    outline: none;
    margin-top: 5px;
    width: 100%;
    height: 30px;
    padding-left: 10px;
    font-size: x-large;

}

.searchList{
    border: 1 solid black;
}

.backgroundImg-welcome{
    background-image: url(../../assets/images/backGroundImg.png);
}

::placeholder{
    color:black;
    opacity: 1;
} */