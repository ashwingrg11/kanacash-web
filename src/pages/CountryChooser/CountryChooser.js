import React, { Component } from "react";
import { Col, Row, Container } from "react-bootstrap";
import { withRouter } from "react-router-dom";
import { connect } from "react-redux";
import star from "../../assets/images/star.svg";
import ROUTES from "../../assets/Routes/Routes";
import {
  getDestination,
  getExchangeRate,
  getAllCountries,
} from "../../store/actions/miscellaneousAction";
import PropTypes from "prop-types";

import Header from "../../containers/layout/Header";
import Footer from "../../containers/layout/Footer";
import { getTransferFee } from "../../store/actions/TrasactionAction";
import { getSenderTransactionLimit } from "../../store/actions/senderDetail";
import { InputDropdown } from "../../components/CustomDropdown";
import "./CountryChooser.css";
import COLORS from "../../assets/Color/Color";
import CustomAlert from "../../components/Alert/CustomAlert";
import Disclaimer from "../../components/Disclaimer/Disclaimer";
class CountryChooser extends Component {
  constructor(props) {
    super(props);
    this.state = {
      selectedCountry: null,
      countryList: [],
      error: "",
      loading: true,
      showAlert: false,
      message: "",
      variant: "",
    };
  }

  componentDidMount = () => {
    localStorage.removeItem("signUpFromTransaction");
    this.setState({ loading: true });
    this.props
      .getAllCountries()
      .then((res) => this.setState({ loading: false }));
    // this.props.getDestination({ sourceCountry: "US" });
    // TODO: missing api endpoints
    // this.props.getTransferFee();
    // this.props.getExchangeRate();
    if (localStorage.accessToken) {
      this.props.getSenderTransactionLimit();
    } else {
      // this.props.clearUser();
    }
    let state = this.props.location.resetPasswordSuccess;
    if (
      this.props.location?.resetPasswordSuccess &&
      localStorage.successReset
    ) {
      this.setState(state);
      localStorage.removeItem("successReset");
    }
  };

  componentDidUpdate = (prevProps) => {
    if (prevProps.countries !== this.props.countries) {
      const countryList = this.props.countries.filter(
        (item) => item.three_char_code !== "USA"
      );
      this.setState({
        countryList: countryList,
      });
    }
  };

  nextStep = () => {
    if (this.state.selectedCountry) {
      localStorage.setItem(
        "selectedCountry",
        JSON.stringify(this.state.selectedCountry)
      );
      const route = {
        pathname: ROUTES.PAYMENT_DETAIL,
        state: this.state.selectedCountry,
      };
      this.props.history.push(route);
    } else {
      this.setState({ error: "Select country you are sending money to." });
    }
  };

  selectCountry = (value) => {
    this.setState({ selectedCountry: value, error: "" });
  };

  // componentDidMount = () => {
  //   console.log("state", this.props.location);
  //   if (this.props.location?.state) {
  //     let state = this.props.location.state;
  //     this.setState(state);
  //   }
  // };

  render() {
    return (
      <React.Fragment>
        <Header />
        <section className="front-background">
          <Container>
            <Row>
              <div className="circle-stroke country-circle">
                <Col>
                  <Row>
                    <Col className="text-center p-0">
                      <h2 className="title">
                        <img src={star} alt="star-icon" className="ml-1" />
                        <span style={{ color: `${COLORS.PRIMARY_COLOR}` }}>
                          {" "}
                          &nbsp; SEND MONEY ANYWHERE &nbsp;{" "}
                        </span>
                        <img src={star} alt="star-icon" className="mr-1" />
                      </h2>
                      <p className="tagline">
                        We make it possible to send money at lowest fees.
                      </p>
                      <div className="calculator">
                        <InputDropdown
                          label={"Send money to"}
                          option={this.state.countryList || []}
                          onClick={() => this.nextStep()}
                          onChangeOption={(value) => this.selectCountry(value)}
                          value={this.state.selectedCountry}
                          loading={this.state.loading}
                        />
                        {this.state.error && (
                          <p
                            style={{ fontSize: ".8rem", width: "100%" }}
                            className="p-2 text-danger mb-1"
                          >
                            {this.state.error}
                          </p>
                        )}
                      </div>
                    </Col>
                  </Row>
                </Col>
              </div>
            </Row>
          </Container>
        </section>
        <Footer />
        {!localStorage.acceptPolicy && <Disclaimer />}
        <CustomAlert
          show={this.state.showAlert}
          message={this.state.message}
          variant={this.state.variant}
          close={() => this.setState({ showAlert: false })}
        />
      </React.Fragment>
    );
  }
}

const mapStateToProps = (state) => ({
  countries: state.miscellaneous.countries,
  acceptPolicy: state.auth.accept_policy,
});

const mapDispatchToProps = {
  getAllCountries,
  getDestination,
  getExchangeRate,
  getTransferFee,
  getSenderTransactionLimit,
};

CountryChooser.propTypes = {
  getDestination: PropTypes.func,
  getTransferFee: PropTypes.func,
  getExchangeRate: PropTypes.func,
  destinationCountry: PropTypes.array,
  history: PropTypes.object,
  getSenderTransactionLimit: PropTypes.func,
  getAllCountries: PropTypes.func,
  countries: PropTypes.array,
  location: PropTypes.object,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withRouter(CountryChooser));
