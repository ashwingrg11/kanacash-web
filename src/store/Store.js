import { createStore, combineReducers, applyMiddleware, compose } from "redux";
// import { composeWithDevTools } from 'redux-devtools-extension';
import Thunk from "redux-thunk";
import LoginReducer from "./reducers/Login";
import ForgotPasswordReducer from "./reducers/ForgotPassword";
import LoaderReducer from "./reducers/Loader";
import ModalReducer from "./reducers/Modal";
import NetworkReducer from "./reducers/Network";
import AuthReducer from "./reducers/Auth";
import ErrorReducer from "./reducers/Error";
import MiscellaneousReducer from "./reducers/MiscellaneousReducer";
import senderDetailReducers from "./reducers/SenderDetailReducer";
import TransactionReducer from "./reducers/TransactionReducer";
import SignUpReducer from "./reducers/SignUpReducer";
import AdminReducers from "./reducers/adminReducers";

const middleware = [Thunk];
const initialState = {};

const composeEnhancers =
  typeof window === "object" && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
        trace: true,
      })
    : compose;

const rootReducer = combineReducers({
  login: LoginReducer,
  forgotPassword: ForgotPasswordReducer,
  loader: LoaderReducer,
  modal: ModalReducer,
  network: NetworkReducer,
  auth: AuthReducer,
  error: ErrorReducer,
  miscellaneous: MiscellaneousReducer,
  senderDetail: senderDetailReducers,
  transaction: TransactionReducer,
  signup: SignUpReducer,
  admin: AdminReducers,
});

const store = createStore(
  rootReducer,
  initialState,
  composeEnhancers(applyMiddleware(...middleware))
);
export default store;
