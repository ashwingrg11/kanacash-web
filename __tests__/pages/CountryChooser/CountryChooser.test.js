import React from "react";
import CountryChooser from "../../../src/pages/CountryChooser/CountryChooser";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import { shallow } from "enzyme";
import store from "../../../src/store/Store";

import renderer from "react-test-renderer";
describe("<CountryChooser />", () => {
  it("CountryChooser renders correctly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <CountryChooser />
          </Router>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
