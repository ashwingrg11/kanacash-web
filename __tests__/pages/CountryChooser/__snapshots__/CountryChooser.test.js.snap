// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CountryChooser /> CountryChooser renders correctly 1`] = `
Array [
  <header
    className="header-main"
  >
    <div
      className="container"
    >
      <nav
        className="navbar navbar-expand-lg navbar-light bg-light"
      >
        <div
          className="navbar-brand"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
          <img
            alt="logo"
            className="logo"
            src=""
          />
        </div>
        <button
          aria-controls="basic-navbar-nav"
          aria-label="Toggle navigation"
          className="navbar-toggler collapsed"
          onClick={[Function]}
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
        <div
          aria-expanded={null}
          className="navbar-collapse collapse"
          id="basic-navbar-nav"
        >
          <div
            className="ml-md-auto navbar-nav"
            onKeyDown={[Function]}
          >
            <span
              className="btn btn-primary btn-theme btn-primary-border"
              onClick={[Function]}
            >
              Sign in
            </span>
            <button
              className="btn btn-primary btn-theme btn-primary-theme"
              onClick={[Function]}
            >
              Sign up
            </button>
          </div>
        </div>
      </nav>
    </div>
  </header>,
  <section
    className="front-background"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="circle-stroke country-circle"
        >
          <div
            className="col"
          >
            <div
              className="row"
            >
              <div
                className="text-center p-0 col"
              >
                <h2
                  className="title"
                >
                  <img
                    alt="star-icon"
                    className="ml-1"
                    src=""
                  />
                  <span
                    style={
                      Object {
                        "color": "#1077BE",
                      }
                    }
                  >
                     
                      SEND MONEY ANYWHERE  
                     
                  </span>
                  <img
                    alt="star-icon"
                    className="mr-1"
                    src=""
                  />
                </h2>
                <p
                  className="tagline"
                >
                  We make it possible to send money at lowest fees.
                </p>
                <div
                  className="calculator"
                >
                  <div
                    className="sc-AxjAm jhiTqR"
                  >
                    <div
                      className="sc-AxirZ fpkyRk input-dropdown"
                    >
                      <div
                        className="sc-AxiKw fwFcZp custom-wrap"
                        onClick={[Function]}
                      >
                        <span
                          className="sc-AxmLO kPXgwU"
                        >
                          Send money to
                        </span>
                        <div
                          className="sc-AxgMl fqhLTf"
                        >
                          <div
                            className="selected-country"
                          >
                            <input
                              className="sc-fzoLsD bZwsmY"
                              onChange={[Function]}
                              placeholder="Select a country"
                              value=""
                            />
                          </div>
                          <img
                            alt="arrow-dwon"
                            src=""
                          />
                        </div>
                      </div>
                    </div>
                    <button
                      className="sc-fzozJi gZqjid btn btn-primary btn-theme"
                      onClick={[Function]}
                    >
                      Send
                    </button>
                  </div>
                  
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>,
  <footer
    className="footer-section text-center"
  >
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-12"
        >
          <div
            className="foot-nav"
          >
            <a
              aria-current="page"
              className="link active"
              href="/"
              onClick={[Function]}
              style={Object {}}
            >
              Home
            </a>
            <a
              aria-current={null}
              className="link"
              href="/about-us"
              onClick={[Function]}
            >
              About Us
            </a>
            <a
              aria-current={null}
              className="link"
              href="/terms-of-service"
              onClick={[Function]}
            >
              Terms
            </a>
            <a
              aria-current={null}
              className="link"
              href="/privacy-policy"
              onClick={[Function]}
            >
              Privacy Policy
            </a>
            <a
              aria-current={null}
              className="link"
              href="/copyright"
              onClick={[Function]}
            >
              Copyright
            </a>
            <a
              aria-current={null}
              className="link"
              href="/contact-us"
              onClick={[Function]}
            >
              Contact
            </a>
          </div>
        </div>
        <div
          className="copyright col-sm-12"
        >
          <p>
            © Golden Money Transfer Inc. 2020. All Rights Reserved.
             
            <b>
              Licensed by
            </b>
             Department of Financial Institutions, Nevada
          </p>
        </div>
      </div>
    </div>
  </footer>,
  <div
    className="disclaimer text-center"
  >
    <div
      className="container"
    >
      <div
        className="policy"
      >
         
        This website uses cookies to ensure you get the best experience on our website, as described in our
         
        <span
          className="btn-link"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
           
            privacy policy.
        </span>
      </div>
      <div
        className="policy1"
      >
        This website uses cookies,
        <span
          className="btn-link"
          onClick={[Function]}
          style={
            Object {
              "cursor": "pointer",
            }
          }
        >
           
            learn more.
        </span>
      </div>
      <div
        className="buttonDiv"
      >
        <button
          className="btn-outline-warning"
          onClick={[Function]}
          style={
            Object {
              "border": "none",
              "borderRadius": "50px",
              "color": "black",
              "padding": "5px 15px",
            }
          }
        >
          Accept
        </button>
      </div>
    </div>
  </div>,
]
`;
